File: go/binary_search/exists_test.go
package binary_search

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestExists_RandomCases(t *testing.T) {
	for range 10_000 {
		td := newRandomTestData()

		expected := td.expectedFirst >= 0
		actual := ExistsInt(td.array, td.val)
		if actual != expected {
			assert.FailNowf(t, "",
				"ExistsInt(%v, %v):\n  Actual: %v\nExpected: %v",
				td.array, td.val, actual, expected,
			)
		}
	}
}

File: python/binary_search/exists_test.py

from bs_random_test_data import RandomTestData
from exists import exists_int


def test_exists_random_cases():
	for _ in range(10_000):
		td = RandomTestData()

		want = td.expected_first >= 0
		got = exists_int(td.array, td.val)
		assert got == want
